code/clue.js

```diff
@@ -1,59 +1,363 @@
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+    if (!killerIsPicked) {
+    document.getElementById('killerLoader').style.visibility='visible' 
+    setTimeout(function () {
+      document.getElementById('killerLoader').style.visibility='hidden'
+      shuffleFavouriteWeapon(suspects); //This randomly selects a favourite weapon.
+      mystery.killer = randomSelector(suspects);
+      document.getElementById('killerCard').style.background = mystery.killer.color;
+      document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+      document.getElementById('killerAge').innerHTML = mystery.killer.age;
+      document.getElementById('killerOccupation').innerHTML = mystery.killer.occupation;
+      document.getElementById('killerImage').src = mystery.killer.image;
+      document.getElementById('killerDescription').innerHTML = mystery.killer.description;
+      document.getElementById('favouriteWeapon').innerHTML = `Favourite weapon: ${mystery.killer.favouriteWeapon}`;
+      killerIsPicked = true;
+      }, 2500);
+    }
+};
+
+document.getElementById('killerCard').addEventListener('click', pickKiller);
+
+// Pick weapon function
+const pickWeapon = () => {
+  if (!weaponIsPicked) {
+    document.getElementById('weaponLoader').style.visibility='visible';
+    setTimeout(function () {
+      document.getElementById('weaponLoader').style.visibility='hidden';
+      mystery.weapon = randomSelector(weapons); 
+      document.getElementById('weaponImage').src = mystery.weapon.image;
+      document.getElementById('weaponName').innerHTML = mystery.weapon.name;
+      document.getElementById('weaponWeight').innerHTML = `${mystery.weapon.weight}kg`;
+      weaponIsPicked = true;
+    }, 2500)
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+};
+document.getElementById('weaponCard').addEventListener('click', pickWeapon);
+
+// Pick room function
+const pickRoom = () => {
+    if (!roomIsPicked) {
+    document.getElementById('roomLoader').style.visibility='visible';
+    setTimeout(function () {
+      document.getElementById('roomLoader').style.visibility='hidden';
+      mystery.room = randomSelector(rooms);
+      document.getElementById('roomName').innerHTML = mystery.room.name;
+      document.getElementById('roomImage').src = mystery.room.image;
+      roomIsPicked = true
+    }, 2500)
+  }
+};
+
+document.getElementById('roomcard').addEventListener('click', pickRoom);
+
+
+// Reveal function
+
+//const revealMystery = () => { - this function does not work when created using const keyword
+  function revealMystery() {
+    if (killerIsPicked === true && weaponIsPicked === true && roomIsPicked ===true) {
+      document.getElementById('mystery').innerHTML = `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room.name} with the ${mystery.weapon.name}`;
```

Comment:
A tip if you would have wanted a message to be presented for the user if you try to click "Reveal the crime" before all deck cards where picked.

```suggestion
  if (
    mystery.killer === '' ||
    mystery.weapon === '' ||
    mystery.room === ''
  ) {
    document.getElementById("mystery").innerHTML =
      "Click all cards above to reveal the crime!";
  } else {
    document.getElementById(
      "mystery"
    ).innerHTML = `The murder was commited by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room} room with a ${mystery.weapon.name}!`;
```